core/sdk/src/extn/client/extn_client.rs

```diff
@@ -240,10 +240,20 @@ impl ExtnClient {
                                 });
                             } else {
                                 // could be main contract
-                                Self::handle_stream(message, self.request_processors.clone());
+                                if !Self::handle_stream(
+                                    message.clone(),
+                                    self.request_processors.clone(),
+                                ) {
+                                    self.handle_no_processor_error(message);
+                                }
                             }
                         } else {
-                            Self::handle_stream(message, self.request_processors.clone());
+                            if !Self::handle_stream(
+                                message.clone(),
```

Comment:
Same here as well. Is it possible to take ownership of the message rather than cloning it?

Reply:
Same as above